core/src/main/java/io/questdb/cairo/wal/WalWriter.java

```diff
@@ -63,7 +62,7 @@ public class WalWriter implements TableWriterAPI {
     private static final long COLUMN_DELETED_NULL_FLAG = Long.MAX_VALUE;
     private final ObjList<MemoryMA> columns;
     private final ObjList<SymbolMapReader> symbolMapReaders = new ObjList<>();
-    private final IntList initialSymbolCounts = new IntList();
+    private final CowIntList initialSymbolCounts;
```

Comment:
we do not remove columns from list, why is there copy-on-write structure here?

Reply:
We don't remove them, but we may add new ones (see `WalMetadataUpdaterBackend#addColumn()`). That's why an `AtomicIntegerArray` won't do here - it assumes a fixed size array.